pkg/auth/oauth/handlers/default_auth_handler.go

```diff
@@ -7,21 +7,23 @@ import (
 	"strings"
 
 	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apiserver/pkg/endpoints/request"
 
 	authapi "github.com/openshift/origin/pkg/auth/api"
 	oauthapi "github.com/openshift/origin/pkg/oauth/apis/oauth"
 )
 
 // unionAuthenticationHandler is an oauth.AuthenticationHandler that muxes multiple challenge handlers and redirect handlers
 type unionAuthenticationHandler struct {
-	challengers      map[string]AuthenticationChallenger
-	redirectors      *AuthenticationRedirectors
-	errorHandler     AuthenticationErrorHandler
-	selectionHandler AuthenticationSelectionHandler
+	challengers          map[string]AuthenticationChallenger
+	redirectors          *AuthenticationRedirectors
+	errorHandler         AuthenticationErrorHandler
+	selectionHandler     AuthenticationSelectionHandler
+	requestContextMapper request.RequestContextMapper
 }
 
 // NewUnionAuthenticationHandler returns an oauth.AuthenticationHandler that muxes multiple challenge handlers and redirect handlers
-func NewUnionAuthenticationHandler(passedChallengers map[string]AuthenticationChallenger, passedRedirectors *AuthenticationRedirectors, errorHandler AuthenticationErrorHandler, selectionHandler AuthenticationSelectionHandler) AuthenticationHandler {
+func NewUnionAuthenticationHandler(passedChallengers map[string]AuthenticationChallenger, passedRedirectors *AuthenticationRedirectors, errorHandler AuthenticationErrorHandler, selectionHandler AuthenticationSelectionHandler, requestContextMapper request.RequestContextMapper) AuthenticationHandler {
```

Comment:
@openshift/sig-security ptal

Reply:
@simo5 @enj The audit wiring here will end up with you.  You ought to take a look.